### PR TITLE
TST: test CPython 3.13 (free-threading flavor)

### PR DESCRIPTION
**Pull request by neutrinoceros**

### CODE REVIEW

**Comment by neutrinoceros** [comment]
I thought numpy dev would build easily already, but maybe I'm doing something wrong. Will re-assess in a couple weeks to months.

**Comment by neutrinoceros** [comment]
Building works but I really wasn't expecting tests to pass too. I need to better understand what's going on before I can confidently merge this.

File: src/gpgi/_lib.pyx [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# cython: freethreading_compatible = True
```
Comment (neutrinoceros) [inline src/gpgi/_lib.pyx:1]: This macro is not documented yet, so I'll wait until https://github.com/cython/cython/pull/6298 is merged before I can call it stable enough

**Comment by neutrinoceros** [comment]
I think the reason why tests pass already is that all loops run in single threads (i.e.,  I use `range` instead of `prange` everywhere), and I'm not using more than one thread anywhere in my test setup. Overall, particle deposition is *extremely* thread-unsafe, but it shouldn't matter in practice as I don't see how users could be affected.

File: src/gpgi/_lib.pyx (hunk 1 above)
Comment (da-woods) [reply src/gpgi/_lib.pyx:1]: It's documented in https://cython.readthedocs.io/en/latest/src/userguide/source_files_and_compilation.html#compiler-directives already.

(With that said, you should the official line is that everything free-threaded is experimental for the foreseeable future and may change as needed. But that particular directive is probably fairly unlikely to change)

Comment (neutrinoceros) [reply src/gpgi/_lib.pyx:1]: thanks, I missed that ! And dully noted, all of this is still experimental, I'm just experimenting along :)

**Comment by neutrinoceros** [comment]
I understand now that I need to *exercise* multi-threading in CI for this to have value beyond just knowing that gpgi builds in 3.13t. I got started on that in #233

**Pull request merged**